internal/util/tls/tls_cert.go

```diff
@@ -0,0 +1,48 @@
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+)
+
+// ExtractClientCertificates extracts tls.Certificates from TLSClientConfig.
+// It returns an empty slice in case there was no client cert and/or client key provided.
```

Comment:
When called like this `tlsConfig.Certificates = append(tlsConfig.Certificates, clientCertificates...)` in `internal/adminapi/kong.go` I feel this isn't as clear as it could be when it would return just a cert instead of a slice.

I would certainly expect to get something when no error was returned. WDYY?

I mean it **will** work